cli/cmd/check.go

```diff
@@ -103,14 +107,21 @@ func configureAndRunChecks(options *checkOptions) {
 
 	success := runChecks(os.Stdout, hc)
 
-	fmt.Println("")
-
 	if !success {
 		fmt.Printf("Status check results are %s\n", failStatus)
 		os.Exit(2)
 	}
 
 	fmt.Printf("Status check results are %s\n", okStatus)
+
+	return nil
+}
+
+func (o *checkOptions) validate() error {
+	if o.preInstallOnly && o.dataPlaneOnly {
+		return fmt.Errorf("--pre and --proxy flags are mutually exclusive")
```

Comment:
though we're not perfectly consistent with this across the codebase, i'd recommend `errors.New()` if you don't need variable substitution.